prod_api/tests/test_auth.py

```diff
@@ -0,0 +1,336 @@
+        resp = superdesk_client.post(
+            url_for('auth_server.issue_token'),
+            data={
+                'grant_type': 'client_credentials'
+            },
+            headers={
+                'Authorization': _basic_auth_str(client_id, password)
+            }
+        )
+
+    # we get an access token
+    resp_data = json.loads(resp.data.decode('utf-8'))
+    token = resp_data['access_token']
+
+    # we drop a superdesk flask app and client to avoid conflict between flask apps
+    teardown_app(superdesk_app)
+    del superdesk_client
+
+    # we create a prodapi flask app and client
+    prodapi_app = get_test_prodapi_app()
+    prodapi_client = prodapi_app.test_client()
+
+    # we change `AUTH_SERVER_SHARED_SECRET`
+    prodapi_app.config['AUTH_SERVER_SHARED_SECRET'] = '7fZOf0VI9T70vU5uNlKLrc5GlabxVgl6'
+
+    # we send a request with an auth token
+    with prodapi_app.test_request_context():
+        for resource in ('archive',):
+            # we send a request with a token
+            resp = prodapi_client.get(
+                url_for('{}|resource'.format(resource)),
+                headers={
+                    'Authorization': 'Bearer {}'.format(token)
+                }
+            )
+            resp_data = json.loads(resp.data.decode('utf-8'))
+
+    # we get a 401 response
+    assert resp.status_code == 401
+    assert resp_data == {
+        '_status': 'ERR',
+        '_error': {
+            'code': 401,
+            'message': 'Please provide proper credentials'
+        }
+    }
+
+
+@pytest.mark.parametrize('issued_tokens', [(SCOPES, tuple())], indirect=True)
+def test_scopes(issued_tokens, prodapi_app, prodapi_client):
+    """
+    Send a request with 'all scopes' token:
+        - make a requests to all endpoints with a token
+        - ensure that all requests are authorized by prod api
+
+    Send a request with 'empty scopes' token:
+        - make a requests to all endpoints with a token
+        - ensure that all requests are NOT authorized by prod api due to empty scopes
+
+    :param issued_tokens: list of tokens provided by `issued_tokens` fixture
+    :type issued_tokens: list
+    :param prodapi_app: prod api app
+    :type prodapi_app: eve.flaskapp.Eve
+    :param prodapi_client: client for prod api
+    :type prodapi_client: flask.testing.FlaskClient
+    """
+
+    access_token_all_scopes = issued_tokens[0]['access_token']
+    access_token_no_scopes = issued_tokens[1]['access_token']
+
+    with prodapi_app.test_request_context():
+        for resource in RESOURCES:
+            # we send a requests with 'empty scopes' token
+            resp = prodapi_client.get(
+                url_for('{}|resource'.format(resource)),
+                headers={
+                    'Authorization': 'Bearer {}'.format(access_token_no_scopes)
+                }
+            )
+            resp_data = json.loads(resp.data.decode('utf-8'))
+            # we get a 401 response
```

Comment:
The sequence diagram at https://superdesk.readthedocs.io/en/latest/auth_server.html#overview says that a `403` must be returned in case of invalid scope (`401` is for invalid credentials), so this must be either changed here or in the diagram.